pkg/http/logging/logging.go

```diff
@@ -0,0 +1,62 @@
+package logging
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/felixge/httpsnoop"
+)
+
+type middleware struct {
+	output io.Writer
+}
+
+func New(opts ...Option) func(http.Handler) http.Handler {
+	m := &middleware{output: os.Stderr}
+
+	for _, opt := range opts {
+		opt(m)
+	}
+
+	return m.commonLogMiddleware
+}
+
+func (m *middleware) commonLogMiddleware(next http.Handler) http.Handler {
+	fn := func(w http.ResponseWriter, r *http.Request) {
+		t := time.Now()
+
+		// Execute the chain of handlers, while capturing HTTP metrics: code, bytes-written, duration
+		metrics := httpsnoop.CaptureMetrics(next, w, r)
+
+		host := r.Header.Get("x-forwarded-for")
+		if host == "" {
+			// r.RemoteAddr contains port, which we want to remove
+			idx := strings.LastIndex(r.RemoteAddr, ":")
+			if idx == -1 {
+				host = r.RemoteAddr
+			} else {
+				host = r.RemoteAddr[:idx]
+			}
+		}
+
+		// Combined log format
```

Comment:
Why are we dealing in combined log instead of JSON? I think JSON would be a better place to be starting since it's much easier to parse in most log aggregators.

Reply:
That is what was being used in other services.  I have no objection to change this to logrus, which can be set to customize the output.

Reply:
I don't have an opinion on this but I think this was probably carried over from k8s-auth-portal where we used combined log format.

Reason then was because our router reports logs in that [format](https://mongodb.splunkcloud.com/en-US/app/search/search?q=search%20index%3Dplatform-prod%20sourcetype%3D%22kube%3Acontainer%3Arouter-public-elb-traefik%22&display.page.search.mode=smart&dispatch.sample_ratio=1&workload_pool=standard_perf&earliest=-30m%40m&latest=now&sid=1646772586.451310), so we used same for [auth-portal](https://mongodb.splunkcloud.com/en-US/app/search/search?q=search%20index%3Dplatform-prod%20sourcetype%3D%22kube%3Acontainer%3Aauth%22&display.page.search.mode=smart&dispatch.sample_ratio=1&workload_pool=standard_perf&earliest=-30m%40m&latest=now&sid=1646772549.451267) to make it easier to view